src/wayland/tablet/tablet_tool.rs

```diff
@@ -0,0 +1,398 @@
+            }
+
+            if tool.capabilitys.distance {
+                wl_tool.capability(zwp_tablet_tool_v2::Capability::Distance);
+            }
+
+            if tool.capabilitys.tilt {
+                wl_tool.capability(zwp_tablet_tool_v2::Capability::Tilt);
+            }
+
+            if tool.capabilitys.slider {
+                wl_tool.capability(zwp_tablet_tool_v2::Capability::Slider);
+            }
+
+            if tool.capabilitys.rotation {
+                wl_tool.capability(zwp_tablet_tool_v2::Capability::Rotation);
+            }
+
+            wl_tool.done();
+            self.inner.borrow_mut().instances.push(wl_tool.deref().clone());
+        }
+    }
+
+    /// Notifify that this tool is focused on a certain surface.
+    ///
+    /// You provide the location of the tool, in the form of:
+    ///
+    /// - The coordinates of the tool in the global compositor space
+    /// - The surface on top of which the tool is, and the coordinates of its
+    ///   origin in the global compositor space.
+    pub fn proximity_in(
+        &self,
+        pos: (f64, f64),
+        focus: (WlSurface, (f64, f64)),
+        tablet: &TabletHandle,
+        time: u32,
+    ) {
+        self.inner.borrow_mut().proximity_in(pos, focus, tablet, time)
+    }
+
+    /// Notifify that this tool has left proximity.
+    pub fn proximity_out(&self, time: u32) {
+        self.inner.borrow_mut().proximity_out(time);
+    }
+
+    /// Notify that the tool moved
+    ///
+    /// You provide the new location of the tool, in the form of:
+    ///
+    /// - The coordinates of the tool in the global compositor space
+    /// - The surface on top of which the tool is, and the coordinates of its
+    ///   origin in the global compositor space (or `None` of the pointer is not
+    ///   on top of a client surface).
+    ///
+    /// This will internally take care of notifying the appropriate client objects
+    /// of proximity_in/proximity_out events.
+    pub fn motion(
+        &self,
+        pos: (f64, f64),
+        focus: Option<(WlSurface, (f64, f64))>,
+        tablet: &TabletHandle,
+        time: u32,
+    ) {
+        self.inner.borrow_mut().motion(pos, focus, tablet, time)
+    }
+
+    /// Queue tool pressure update
+    pub fn pressure(&self, pressure: f64) {
+        self.inner.borrow_mut().pressure(pressure);
+    }
+
+    /// Queue tool distance update
+    pub fn distance(&self, distance: f64) {
+        self.inner.borrow_mut().distance(distance);
+    }
+
+    /// Queue tool tilt update
+    pub fn tilt(&self, tilt: (f64, f64)) {
+        self.inner.borrow_mut().tilt(tilt);
+    }
+
+    /// Queue tool slider update
+    pub fn slider_position(&self, slider: f64) {
+        self.inner.borrow_mut().slider_position(slider);
+    }
+
+    /// Queue tool rotation update
+    pub fn rotation(&self, rotation: f64) {
+        self.inner.borrow_mut().rotation(rotation);
+    }
```

Comment:
These methods are documented to queue an update, they should also document when the update will be actually sent to the client.